tests/testthat/test-query_dataset.R

```diff
@@ -15,3 +15,45 @@ test_that("No indicator supplied", {
     query_dataset(example_id())
   )
 })
+
+test_that("Geography query returns expected geographies", {
+  expect_equal(
+    post_dataset(
+      example_id(group = "attendance"),
+      indicators = example_id("indicator", group = "attendance"),
+      time_periods = "2024|W23",
+      geographies = c("NAT|id|dP0Zw", "REG|id|rg3Nj"),
+      filter_items = c("pmRSo")
+    ) |>
+      dplyr::select(geographic_level, NAT, REG) |>
+      dplyr::distinct(),
+    data.frame(
+      geographic_level = c("NAT", "REG"),
+      NAT = c("dP0Zw :: England (code = E92000001)", "dP0Zw :: England (code = E92000001)"),
+      REG = c(NA, "rg3Nj :: East Midlands (code = E12000004)")
+    )
+  )
+})
+
+test_that("Test filter-combinations POST dataset query", {
+  query_result <- query_dataset(
+    example_id(group = "attendance"),
+    indicators = example_id("indicator", group = "attendance"),
+    time_periods = "2024|W23",
+    geographies = c("NAT|id|dP0Zw", "REG|id|rg3Nj"),
+    filter_items = list(
+      attendance_status = c("pmRSo", "7SdXo"),
```

Comment:
The names aren't actually used anywhere. The sqids are unique across all filter items in a given data set, so it doesn't need to know which filter a given item belongs to. The main object above is just to have an organised way for the user to say "a and b are in the same filter and x and y are in another filter". A user could provide something in the form `list(c("a", "b", "c"), c("x", "y"))` and it would still work, but I thought adding the naming to the example list made it clearer how I'm expecting people to use it.